zap/src/parser/convert.rs

```diff
@@ -0,0 +1,336 @@
+use std::collections::HashSet;
+
+use codespan_reporting::diagnostic::{Diagnostic, Label};
+
+use crate::config::{Casing, Config, Enum, EvDecl, Range, Struct, Ty, TyDecl};
+
+use super::syntax_tree::{
+	Spanned, SyntaxBoolLit, SyntaxConfig, SyntaxDecl, SyntaxEnum, SyntaxEnumKind, SyntaxEvDecl, SyntaxIdentifier,
+	SyntaxNumLit, SyntaxOptValueKind, SyntaxRange, SyntaxRangeKind, SyntaxStrLit, SyntaxStruct, SyntaxTy, SyntaxTyDecl,
+	SyntaxTyKind,
+};
+
+pub fn convert(syntax_config: SyntaxConfig<'_>) -> (Config<'_>, Vec<Diagnostic<()>>) {
+	let mut state = ConvertState::new(
+		syntax_config
+			.decls
+			.iter()
+			.filter(|decl| matches!(decl, SyntaxDecl::Ty(_)))
+			.map(|decl| match decl {
+				SyntaxDecl::Ty(tydecl) => tydecl.name.name,
+				_ => unreachable!(),
+			})
+			.collect(),
+	);
+
+	let config = syntax_config.into_config(&mut state);
+
+	(config, state.into_daigs())
+}
+
+struct ConvertState<'src> {
+	daigs: Vec<Diagnostic<()>>,
```

Comment:
This name is... interesting

Reply:
I unfortunately can't think of a better one

Reply:
Don’t shorten it - `diagnostics`